apps/native/app/src/screens/document-detail/utils/share-file.tsx

```diff
@@ -1,30 +1,56 @@
 import Share from 'react-native-share'
+import RNFS from 'react-native-fs'
 
 import { isAndroid } from '../../../utils/devices'
 import { authStore } from '../../../stores/auth-store'
 import { DocumentV2 } from '../../../graphql/types/schema'
 
 interface ShareFileProps {
   document: DocumentV2
+  type: 'pdf' | 'html' | 'url'
   pdfUrl?: string
+  content?: string | null
 }
 
-export const shareFile = async ({ document, pdfUrl }: ShareFileProps) => {
+export const shareFile = async ({
+  document,
+  pdfUrl,
+  type,
+  content,
+}: ShareFileProps) => {
   if (!document || !document.subject || !document.sender) {
     return
   }
+  const isHtml = type === 'html'
+  let htmlUrl: string | undefined
 
   if (isAndroid) {
     authStore.setState({ noLockScreenUntilNextAppStateActive: true })
   }
 
+  if (isHtml) {
+    const utf8Prefix = '<meta charset="UTF-8">'
+    const formattedHtml = utf8Prefix + content
+    try {
+      const path = `${RNFS.TemporaryDirectoryPath}/${document.subject}.html`
+      await RNFS.writeFile(path, formattedHtml, 'utf8')
+      htmlUrl = path
+    } catch (error) {
+      // noop
```

Comment:
I'm thinking if we should add error message if this or the sharing fails? Just to let the user know that it did not work. Toast (when we have added that globally) for example

Reply:
Yeah. In the meantime we may want to log the error in datadog and return?

Reply:
Good idea! But I don't think we are logging errors in datadog in any place in the app 😅 At least I have never done it, need to look into it